### PR TITLE
Fix failing deployments

### LINKED ISSUE #1215: Some deployments are not going through due to unmet requirements
**What were you trying to achieve?**
Deploy api's to staging & production

**What are the expected results?**
All deployments should be successful

**What are the received results?**
Building of some images was failing

**What are the steps to reproduce the issue?**
Try to deploy (data-mgt api for example) to staging using the cicd pipeline

**Additional context**
- image
- image

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- [x] [add missing dependecy: image-tag](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/4e498a997b58f829d622b623e6b8500f3c2994bf)
- [x] [correct kafka-connector path](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/b8eee9e901fb8fe8d16290115e752bf18b1bd176)
- [x] [correct meta-data registry variables](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/8179126c062bdcc939c92dd1da108b4b57a77e83)
- [x] [reference right channel data image](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/52e567ec81310232a91a6237a786a951d8ef9042)
- [x] [update the build push action version](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/422ecde7829e79c9be18d84d0695a786aec6e00b)
- [x] [remove deprecated use of 'updateFile'](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/8be3429e09829236c151d2881baef930f14b0220)
- [x] [trigger all workflows to run](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/1783dca81c59c59d5fd004467a512085794fc5f5)
- [x] [update docker login action version](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/055eab7de17ed072a99813b0ce1354f84ec14d59)
- [x] [update k8s-context action version](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/cf72bd177d9aa68dc4f741cd700df1be326ff1c7)
- [x] [update files with retry](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/9ded1a6e12446b99e612be513e35166b9edcb62f)

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- Closes #1215 

**_HOW DO I TEST OUT THIS PR?_**
- Checkout the test run workflow [here](https://github.com/airqo-platform/AirQo-api/actions/runs/3638348762)
- Note that the Kafka-connectors job is still in development cc @NoahNsimbe 
- image

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_**
- n/a

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- n/a

**_ARE THERE ANY RELATED PRs?_**
- n/a

### CODE REVIEW

File: .github/workflows/deploy-apis-to-production.yml [hunk 1]
```diff
@@ -341,10 +341,10 @@ jobs:
           password: ${{ secrets.GCR_CONFIG }}
 
       - name: Build and push measurements source connectors
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3
         with:
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:345]: [update the build push action version](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/422ecde7829e79c9be18d84d0695a786aec6e00b)

File: .github/workflows/deploy-apis-to-production.yml [hunk 2]
```diff
@@ -341,10 +341,10 @@ jobs:
...
           push: true
-          context: src/kafka/connectors/measurements-source-connector
+          context: src/kafka-connectors/measurements-source-connector
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:347]: [correct kafka-connector path](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/b8eee9e901fb8fe8d16290115e752bf18b1bd176)

File: workflow-trigger [hunk 3]
```diff
@@ -1 +1 @@
-2022-12-02T20:00:00Z
+2022-12-07T20:00:00Z
```
Comment (123MwanjeMike) [inline workflow-trigger:1]: [trigger all workflows to run](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/1783dca81c59c59d5fd004467a512085794fc5f5)

File: .github/workflows/deploy-apis-to-production.yml [hunk 4]
```diff
@@ -673,7 +670,6 @@ jobs:
           valueFile: "k8s/device-monitor/values-prod.yaml"
           propertyPath: "image.tag"
           value: ${{ needs.image-tag.outputs.build_id }}
-          updateFile: true
           branch: ${{ env.DEPLOY_BRANCH }}
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:673]: [remove deprecated use of 'updateFile'](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/8be3429e09829236c151d2881baef930f14b0220)

File: .github/workflows/deploy-apis-to-production.yml [hunk 5]
```diff
@@ -700,7 +696,7 @@ jobs:
   ### data-mgt ###
   data-mgt:
     name: build-push-deploy-data-mgt
-    needs: check
+    needs: [check, image-tag]
     if: needs.check.outputs.run_data_mgt == 'true'
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:700]: [add missing dependecy: image-tag](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/4e498a997b58f829d622b623e6b8500f3c2994bf)

File: .github/workflows/deploy-apis-to-production.yml [hunk 6]
```diff
@@ -1228,28 +1216,27 @@ jobs:
...
 
       - name: Build and push  API
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3
         with:
           push: true
           context: src/meta-data
           target: api
-          tags: ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-meta-data-api:${{ needs.image-tag.outputs.build_id }},${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-meta-data-api:latest
+          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-meta-data-api:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-meta-data-api:latest
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1225]: [correct meta-data registry variables](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/8179126c062bdcc939c92dd1da108b4b57a77e83)

File: .github/workflows/deploy-apis-to-production.yml [hunk 7]
```diff
@@ -1768,7 +1748,7 @@ jobs:
       - name: Build and Push Docker Image
         run: |
           cd src/channel-data/
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1750]: [reference right channel data image](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/52e567ec81310232a91a6237a786a951d8ef9042)

**Review by 123MwanjeMike — COMMENTED** [review]
Guiding comments

File: .github/workflows/deploy-apis-to-production.yml [hunk 8]
```diff
@@ -371,7 +371,7 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Login to GCR
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2
         with:
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:375]: [update docker login action version](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/055eab7de17ed072a99813b0ce1354f84ec14d59)

File: .github/workflows/deploy-apis-to-staging.yml [hunk 9]
```diff
@@ -1140,12 +1117,12 @@ jobs:
 
       - name: Deploy to K8S
         run: |
-          kubectl apply -f k8s/device-status/stage-airqo-stage-device-status-job.yaml
+          kubectl apply -f k8s/device-status/stage-airqo-device-status-job.yaml
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-staging.yml:1121]: [rectify device-status typo](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/a4894b0ba56f6216b6b3f78c2d0174d5eac98aec)

File: .github/workflows/deploy-apis-to-production.yml [hunk 10]
```diff
@@ -334,21 +334,21 @@ jobs:
...
           push: true
-          context: src/kafka/connectors/measurements-source-connector
+          context: src/kafka-connectors/measurements-source-connector
           tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/measurements-connect:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/measurements-connect:latest
 
       - name: Login to k8s cluster
-        uses: azure/k8s-set-context@v1
+        uses: azure/k8s-set-context@v2
         with:
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:352]: [update k8s-context action version](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/cf72bd177d9aa68dc4f741cd700df1be326ff1c7)

**Review by Baalmart — COMMENTED** [review]
Hi @123MwanjeMike ,  thanks for this, This is very much welcome. Deployment failures need to be addressed in a timely manner.

On a side note, please add the `work in progress` as an additional tag/label accordingly. It close to impossible to notice the draft status of the PR.

**Comment by 123MwanjeMike** [comment]
> Hi @123MwanjeMike , thanks for this, This is very much welcome. Deployment failures need to be addressed in a timely manner.
> 
> On a side note, please add the `work in progress` as an additional tag/label accordingly. It close to impossible to notice the draft status of the PR.

Right away @Baalmart

File: .github/workflows/deploy-apis-to-production.yml [hunk 11]
```diff
@@ -786,13 +796,15 @@ jobs:
...
-          value: ${{ needs.image-tag.outputs.build_id }}
-          branch: ${{ env.DEPLOY_BRANCH }}
-          message: "Update analytics production image tag to ${{ needs.image-tag.outputs.build_id }}"
+          action: fjogeleit/yaml-update-action@main
+          with: |
+            valueFile: "k8s/analytics/api/values-prod.yaml"
+            propertyPath: "image.tag"
+            value: ${{ needs.image-tag.outputs.build_id }}
+            branch: ${{ env.DEPLOY_BRANCH }}
+            message: "Update analytics production image tag to ${{ needs.image-tag.outputs.build_id }}"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:807]: [update files with retry](https://github.com/airqo-platform/AirQo-api/pull/1216/commits/9ded1a6e12446b99e612be513e35166b9edcb62f)

File: .github/workflows/deploy-apis-to-staging.yml [hunk 12]
```diff
@@ -1363,43 +1366,44 @@ jobs:
...
 
       - name: Build and push  API
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3
         with:
           push: true
           context: src/meta-data
           target: api
-          tags: ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-meta-data-api:${{ needs.image-tag.outputs.build_id }},${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-meta-data-api:latest
+          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-stage-meta-data-api:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-stage-meta-data-api:latest
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-staging.yml:1388]: These are the changes that correct the failing auth service deployment @Baalmart

**Comment by 123MwanjeMike** [comment]
> This LGTM, thanks @123MwanjeMike ...
> 
> A quick one....I recently realised that we have some deployments to production for device-registry regarding the `.envs` are not working as expected?
> 
> PR in question is [this one](https://github.com/airqo-platform/AirQo-api/pull/1210), already merged to production. Related issues? While reviewing [that PR](https://github.com/airqo-platform/AirQo-api/pull/1210), please pay attention to the PR description to appreciate the context a little better.

Hi @Baalmart, in regard to the failing auth-service deployment, this PR will fix it given that the bcc PR has merged before the (envs/secrets) issue was resolved. The exact changes that fix that are [here](https://github.com/airqo-platform/AirQo-api/pull/1216#discussion_r1045873201)

**Review by NoahNsimbe — APPROVED** [review]
LGTM

File: .github/workflows/deploy-apis-to-staging.yml (hunk 12 above)
Comment (123MwanjeMike) [reply .github/workflows/deploy-apis-to-staging.yml:1388]: That won't be necessary @Baalmart. Merging this PR automatically handles that

Comment (123MwanjeMike) [reply .github/workflows/deploy-apis-to-staging.yml:1388]: https://github.com/airqo-platform/AirQo-api/actions/runs/3684274688/workflow

**Comment by 123MwanjeMike** [comment]
https://github.com/airqo-platform/AirQo-api/actions/runs/3684274688/workflow

**Pull request merged**